src/storage/FileResourceStore.ts

```diff
@@ -79,71 +68,70 @@ export class FileResourceStore implements ResourceStore {
+   */
   public async deleteResource(identifier: ResourceIdentifier): Promise<void> {
     let path = this.parseIdentifier(identifier);
     if (path === '' || ensureTrailingSlash(path) === '/') {
-      throw new MethodNotAllowedHttpError('Cannot delete root container.');
+      throw new MethodNotAllowedHttpError('Cannot delete rootFilepath container.');
     }
-    path = `${this.root}${path}`;
-    return new Promise(async(resolve, reject): Promise<any> => {
-      try {
-        const stats = await fsPromises.lstat(path);
-        if (stats.isFile()) {
-          await fsPromises.unlink(path);
-          try {
-            await fsPromises.unlink(`${path}.metadata`);
-          } catch (_) {
-            // It's ok if there was no metadata file.
-          }
-          resolve();
-        } else if (stats.isDirectory()) {
-          path = ensureTrailingSlash(path);
-          const files = await fsPromises.readdir(path);
-          let match = files.find((file): any => !file.startsWith('.'));
-          if (match !== undefined) {
-            reject(new ConflictHttpError('Container is not empty.'));
-          }
-
-          match = files.find((file): any => file.startsWith('.'));
-          while (match) {
-            await fsPromises.unlink(`${path}${match}`);
-            const matchedFile = match;
-            files.filter((file): any => file !== matchedFile);
-            match = files.find((file): any => file.startsWith('.'));
-          }
+    path = `${this.rootFilepath}${path}`;
+    try {
+      const stats = await fsPromises.lstat(path);
+      if (stats.isFile()) {
+        await fsPromises.unlink(path);
+        try {
+          await fsPromises.unlink(`${path}.metadata`);
+        } catch (_) {
+          // It's ok if there was no metadata file.
+        }
+      } else if (stats.isDirectory()) {
+        path = ensureTrailingSlash(path);
+        const files = await fsPromises.readdir(path);
+        let match = files.find((file): any => !file.startsWith('.metadata'));
+        if (match !== undefined) {
+          throw new ConflictHttpError('Container is not empty.');
+        }
 
-          await fsPromises.rmdir(path);
-          resolve();
-        } else {
-          reject(new NotFoundHttpError());
+        match = files.find((file): any => file.startsWith('.metadata'));
+        while (match) {
+          await fsPromises.unlink(`${path}${match}`);
+          const matchedFile = match;
+          files.filter((file): any => file !== matchedFile);
+          match = files.find((file): any => file.startsWith('.metadata'));
         }
-      } catch (error) {
-        reject(new NotFoundHttpError());
+
+        await fsPromises.rmdir(path);
+      } else {
+        throw new NotFoundHttpError();
       }
-    });
+    } catch (error) {
+      if (error instanceof ConflictHttpError) {
+        throw new ConflictHttpError('Container is not empty.');
+      }
+      throw new NotFoundHttpError();
```

Comment:
I'm not a big fan of putting an entire try/catch around this block of code and throwing this error if there is a problem, since this hides potential problems in the code if something else besides checking if the file exists goes wrong. If I'm not mistaken this error should only be thrown if the first `lstat` call throws an error? In all other cases I would not catch the cases since those would imply something went wrong internally with the server.

Reply:
Should an error not also be caught at `unlink(path)`, `readdir(path)`, `rmdir(path)` if this path has already been removed in the meantime. Or don't you see this happening directly?

Reply:
I think having an internal error would be more correct since that would mean something else is modifying those files at the same time which should definitely not happen.

Reply:
Yeah, we have a locking mechanism around it that should take care of this.